src/util/std_expr.h

```diff
@@ -135,6 +135,16 @@ class symbol_exprt:public exprt
   }
 };
 
+namespace std {
```

Comment:
Rather than adding this to the `std` namespace would it be better to define this in a similar way to `irep_id_hash` (i.e., not in the `std` namespace) and just pass it as an extra template parameter when constructing the sharing_maps ?